File: Code/C_Code/02.1.1_ButtonLED/ButtonLED.c
/**********************************************************************
* Filename    : ButtonLED.c
* Description : Controlling an led by button.
* Author      : freenove
* modification: 2016/06/12
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>

#define ledPin    0  	//define the ledPin
#define buttonPin 1		//define the buttonPin

int main(void)
{
	if(wiringPiSetup() == -1){ //when initialize wiring failed,print messageto screen
		printf("setup wiringPi failed !");
		return 1; 
	}
	
	pinMode(ledPin, OUTPUT); //Set ledPin output
	pinMode(buttonPin, INPUT);//Set buttonPin input

	pullUpDnControl(buttonPin, PUD_UP);  //pull up to high level
	while(1){
		
		if(digitalRead(buttonPin) == LOW){ //button has pressed down
			digitalWrite(ledPin, HIGH);   //led on
			printf("led on...\n");
		}
		else {				//button has released 
			digitalWrite(ledPin, LOW);   //led off
			printf("...led off\n");
		}
	}

	return 0;
}

File: Code/Python_Code/02.1.1_ButtonLED/ButtonLED.py
#!/usr/bin/env python3
########################################################################
# Filename    : ButtonLED.py
# Description : Control led with button
# auther      : www.freenove.com
# modification: 2019/12/28
########################################################################
import RPi.GPIO as GPIO

ledPin = 11    # define ledPin
buttonPin = 12    # define buttonPin

def setup():
    
    GPIO.setmode(GPIO.BOARD)      # use PHYSICAL GPIO Numbering
    GPIO.setup(ledPin, GPIO.OUT)   # set ledPin to OUTPUT mode
    GPIO.setup(buttonPin, GPIO.IN, pull_up_down=GPIO.PUD_UP)    # set buttonPin to PULL UP INPUT mode

def loop():
    while True:
        if GPIO.input(buttonPin)==GPIO.LOW: # if button is pressed
            GPIO.output(ledPin,GPIO.HIGH)   # turn on led
            print ('led turned on >>>')     # print information on terminal
        else : # if button is relessed
            GPIO.output(ledPin,GPIO.LOW) # turn off led 
            print ('led turned off <<<')    

def destroy():
    GPIO.output(ledPin, GPIO.LOW)     # turn off led 
    GPIO.cleanup()                    # Release GPIO resource

if __name__ == '__main__':     # Program entrance
    print ('Program is starting...')
    setup()
    try:
        loop()
    except KeyboardInterrupt:  # Press ctrl-c to end the program.
        destroy()

